src/shared/options/readOptions.ts

```diff
@@ -148,3 +165,68 @@ export async function readOptions(args: string[]): Promise<OptionsParseResult> {
 		options: augmentedOptions,
 	};
 }
+
+export const optionsSchema = z
+	.object({
+		author: z.string().optional(),
+		base: z
+			.union([
+				z.literal("everything"),
+				z.literal("minimum"),
+				z.literal("prompt"),
+			])
+			.optional(),
+		createRepository: z.boolean().optional(),
+		description: z.string().optional(),
+		email: z.string().email().optional(),
+		excludeCompliance: z.boolean().optional(),
+		excludeContributors: z.boolean().optional(),
+		excludeLintJson: z.boolean().optional(),
+		excludeLintKnip: z.boolean().optional(),
+		excludeLintMd: z.boolean().optional(),
+		excludeLintPackageJson: z.boolean().optional(),
+		excludeLintPackages: z.boolean().optional(),
+		excludeLintPerfectionist: z.boolean().optional(),
+		excludeLintSpelling: z.boolean().optional(),
+		excludeLintYml: z.boolean().optional(),
+		excludeReleases: z.boolean().optional(),
+		excludeRenovate: z.boolean().optional(),
+		excludeTests: z.boolean().optional(),
+		funding: z.string().optional(),
+		owner: z.string().optional(),
+		repository: z.string().optional(),
+		skipGitHubApi: z.boolean(),
+		skipInstall: z.boolean(),
+		skipRemoval: z.boolean(),
+		skipRestore: z.boolean().optional(),
+		skipUninstall: z.boolean(),
+		title: z.string().optional(),
+	})
+	// This is necessary since we want to require schema fields, but allow values to be undefined
+	// https://stackoverflow.com/questions/71477015/specify-a-zod-schema-with-a-non-optional-but-possibly-undefined-field
+	.transform((o) => ({
```

Comment:
Wasn't sure how to deal with zod making optional fields out of

```ts
z.union([z.string(), z.undefined()]);
```

I assume we don't want to allow non-existent fields while parsing `mappedOptions`.

Reply:
You know, now that we're using Zod I don't mind the fields being optional. It's been inconvenient having to make full objects in tests. I think switching all the `| undefined` properties in `interface Options` to be `?` optional would be the cleanest solution here.